### PR TITLE
Move tfplan* files and output.json file to .terraform folder

### PR DESCRIPTION
**Pull request by kobrikx**
#### What's new: 

 - Move tfplan* files to $ENV_DIR/.terraform folder
 - Move output.json to $ENV_DIR/.terraform folder

### CODE REVIEW

**Comment by kobrikx** [comment]
[Create output.txt and other non-important files in .terraform dir](https://trello.com/c/CYbT5lTD/188-create-outputtxt-and-other-non-important-files-in-terraform-dir)

File: terraform/main.mk [hunk 1]
```diff
@@ -87,9 +87,9 @@ terraform.13upgrade:
 	@ $(CMD_TERRAFORM_MODULES_UPGRADE)
 
 terraform.apply: terraform.plan ## Deploy infrastructure
-	@ cd $(ENV_DIR) && \
-	$(TERRAFORM) apply -input=false tfplan && \
-	$(TERRAFORM) output -json > output.json	&& \
+	@ cd $(ENV_DIR)/.terraform && \
```
Comment (AutomationD) [inline terraform/main.mk:90]: Please explain why are we changing terraform root to .terraform, can't we point to the file without cd-ing into the .terraform?

**Review by AutomationD — CHANGES_REQUESTED** [review]

File: terraform/main.mk (hunk 1 above)
Comment (kobrikx) [reply terraform/main.mk]: We can. Fixed this, thank you.

File: terraform/templates/terraform.tfplan.gotmpl [hunk 2]
```diff
@@ -5,7 +5,7 @@
 <details><summary>Expand to view details</summary><p>
 
 ```
-{{file.Read "/temp/tfplan.txt"}}
+{{file.Read "/temp/.terraform/tfplan.txt"}}
```
Comment (gloovers) [inline terraform/templates/terraform.tfplan.gotmpl:8]: @kobrikx why do we need to change there?

File: terraform/main.mk [hunk 3]
```diff
@@ -124,14 +124,14 @@ terraform.destroy-quiet: ## Destroy infrastructure without confirmation
 	@ echo "\n\033[36m[INFO] Please run: make secrets.delete or app.delete_secrets now\033[0m"
 
 terraform.output-to-ssm: ## Manual upload output.json to AWS SSM. Output.json encoded in base64.
-	@ cd $(ENV_DIR) && \
+	@ cd $(ENV_DIR)/.terraform && \
```
Comment (AutomationD) [inline terraform/main.mk:127]: @kobrikx do we also need to cd into .terraform here? (and why does it work here https://github.com/hazelops/icmk/pull/132/files#diff-d0414a57cfaec3da463ae028b8d3aa467a2f15a09677a49356ba933c58430138R89?)

File: terraform/templates/terraform.tfplan.gotmpl (hunk 2 above)
Comment (kobrikx) [reply terraform/templates/terraform.tfplan.gotmpl:8]: Otherwise it can't find the file.

File: terraform/main.mk (hunk 3 above)
Comment (kobrikx) [reply terraform/main.mk]: Fixed

**Comment by kobrikx** [comment]
[Safely remove output.json generation from ICMK](https://trello.com/c/qnCx5T2H/189-safely-remove-outputjson-generation-from-icmk)

File: terraform/templates/terraform.tfplan.gotmpl [hunk 4]
```diff
@@ -5,7 +5,7 @@
 <details><summary>Expand to view details</summary><p>
 
 ```
-{{file.Read "/temp/tfplan.txt"}}
```
Comment (gloovers) [inline terraform/templates/terraform.tfplan.gotmpl:8]: @kobrikx why do we need to make changes there?

**Review by gloovers — APPROVED** [review]
LGTM

File: terraform/templates/terraform.tfplan.gotmpl (hunk 4 above)
Comment (kobrikx) [reply terraform/templates/terraform.tfplan.gotmpl:8]: Otherwise it can't find the file. This folder is mounted to Gomplate in docker.

**Review by AutomationD — APPROVED** [review]
Looks good

**Pull request merged**